src/storage/src/hummock/store/version.rs

```diff
@@ -819,4 +803,163 @@ impl HummockVersionReader {
+                            .peek()
+                            .is_some()
+                        {
+                            return Ok(false);
+                        }
+                    }
+                    LevelType::Nonoverlapping => {
+                        if prune_nonoverlapping_ssts(&level.table_infos, &encoded_user_key_range)
+                            .peekable()
+                            .peek()
+                            .is_some()
+                        {
+                            return Ok(false);
+                        }
+                    }
+                }
+            }
+            return Ok(true);
+        };
+
+        // 2. order guarantee: imm -> sst
+        for local_sst in &uncommitted_ssts {
+            table_counts += 1;
+            if hit_sstable_bloom_filter(
+                self.sstable_store
+                    .sstable(local_sst, &mut local_stats)
+                    .await?
+                    .value(),
+                bloom_filter_prefix_hash,
+                &mut local_stats,
+            ) {
+                local_stats.report_bloom_filter_metrics(
+                    self.state_store_metrics.as_ref(),
+                    "surely_not_have",
+                    table_id_label,
+                    false,
+                );
+                local_stats.report(self.state_store_metrics.as_ref(), table_id_label);
+                return Ok(false);
+            }
+        }
+
+        // 3. read from committed_version sst file
+        // Because SST meta records encoded key range,
+        // the filter key needs to be encoded as well.
+        assert!(committed_version.is_valid());
+        for level in committed_version.levels(table_id) {
+            if level.table_infos.is_empty() {
+                continue;
+            }
+            match level.level_type() {
+                LevelType::Overlapping | LevelType::Unspecified => {
+                    let sstable_infos =
+                        prune_overlapping_ssts(&level.table_infos, table_id, &table_key_range);
+                    for sstable_info in sstable_infos {
+                        table_counts += 1;
+                        if hit_sstable_bloom_filter(
+                            self.sstable_store
+                                .sstable(sstable_info, &mut local_stats)
+                                .await?
+                                .value(),
+                            bloom_filter_prefix_hash,
+                            &mut local_stats,
+                        ) {
+                            local_stats.report_bloom_filter_metrics(
+                                self.state_store_metrics.as_ref(),
+                                "surely_not_have",
+                                table_id_label,
+                                false,
+                            );
+                            local_stats.report(self.state_store_metrics.as_ref(), table_id_label);
```

Comment:
Can we wrap the `local_stats` with a guard that stores the `elf.state_store_metrics' and `table_id_label` so that we don't have to pay attention to manually reporting before returning. As long as we impl `Deref` and `DerefMut` for the guard, we can use the guard the same way we use the `LocalStats`.

Reply:
For `report_bloom_filter_metrics`, different parameters are returned in different branches, I am not sure if it is easy to modify, we can discuss it in the next pr